From 9d0c4b3d52464cb38b531e610a71ff1e0f6fcf61 Mon Sep 17 00:00:00 2001
From: Bandan Das <bsd@redhat.com>
Date: Wed, 6 Nov 2013 02:11:03 +0100
Subject: [PATCH 56/81] kvmvapic: Clear also physical ROM address when entering INACTIVE state

RH-Author: Bandan Das <bsd@redhat.com>
Message-id: <1383703863-4840-3-git-send-email-bsd@redhat.com>
Patchwork-id: 55473
O-Subject: [PATCH RHEL-7.0 qemu-kvm 3/3] kvmvapic: Clear also physical ROM address when entering INACTIVE state
Bugzilla: 920021
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

To avoid misinterpreting INACTIVE after migration as old qemu-kvm's
STANDBY, also clear rom_state_paddr when going back to this state.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 4357930b8a7d2fcff2d8121ec518117428a781e7)
---
 hw/i386/kvmvapic.c | 2 ++
 1 file changed, 2 insertions(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/kvmvapic.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/hw/i386/kvmvapic.c b/hw/i386/kvmvapic.c
index a96ea85..577ae64 100644
--- a/hw/i386/kvmvapic.c
+++ b/hw/i386/kvmvapic.c
@@ -505,6 +505,7 @@ static void vapic_reset(DeviceState *dev)
     VAPICROMState *s = DO_UPCAST(VAPICROMState, busdev.qdev, dev);
 
     s->state = VAPIC_INACTIVE;
+    s->rom_state_paddr = 0;
     vapic_enable_tpr_reporting(false);
 }
 
@@ -655,6 +656,7 @@ static void vapic_write(void *opaque, hwaddr addr, uint64_t data,
         }
         if (vapic_prepare(s) < 0) {
             s->state = VAPIC_INACTIVE;
+            s->rom_state_paddr = 0;
             break;
         }
         break;
-- 
1.7.1

